### ISSUE TITLE
Tweak the variables in the model to optimize profomance

### ISSUE THREAD

**Issue by LoganWatson1400** [LoganWatson1400/Brady_Hackathon#29]
## Notes
- This is open to anyone who wants to experiment with the model! :D
- The model has lots of overfitting protection, so you don’t need to worry too much about that.
- Avoid forcing the model to run longer than it wants to (it will stop when it's ready). Running it longer may lead to overfitting.
- The model saves backup checkpoints every 10 epochs. However, I haven’t tested them much, so they may not work correctly.
- Comment on this issue your best run with the public branch link :)

---

## cnn_train
```
epochs = 1000
batch_size = 64
patience = 10
```

---

## data_process
```
IMAGE_SIZE_LIMIT = 224
```

---

## make_model
- Learning Rate (`lr`): `0.0001`
- Dropout Rate (`dropout`): `0.5`
- You can experiment with adding more layers, such as:
  ```python
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(dropout),  # Reduce overfitting
  ```
- You can also experiment with the model’s compiler (but that's a bit more advanced). Good luck! Example:
  ```python
  model.compile(optimizer=tf.keras.optimizers.Adam(learning_rate=lr), 
                loss="categorical_crossentropy", 
                metrics=["accuracy"])
  ```

---

## Data Augmentation
- You can use the following augmentation pipeline to enhance the model’s robustness:
  ```python
  data_augmentation = tf.keras.Sequential([
    tf.keras.layers.RandomFlip("horizontal"),
    tf.keras.layers.RandomFlip("vertical"),
    tf.keras.layers.RandomZoom(0.2),
    tf.keras.layers.RandomTranslation(0.05, 0.05),
    tf.keras.layers.RandomCrop(224, 224),
    tf.keras.layers.RandomHeight(0.1),
    tf.keras.layers.RandomWidth(0.1),
    tf.keras.layers.RandomContrast(0.5),  # Further reduced contrast range
    tf.keras.layers.RandomHue(0.2),       # Further reduced hue range
  ])
  ```
- To visualize augmented images, uncomment the line:
  ```python
  # visualize_augmentation(data_augmentation, x_train)
  ```

---

**Comment by LoganWatson1400** [comment]
# Base Stats :)
### Classification Report

```
                                           precision    recall  f1-score   support

                          No Violation       0.77      1.00      0.87        37
             ANSI A13.1 (Pipe Marking)       1.00      1.00      1.00         5
ANSI Z358.1-2014 (Emergency Equipment)       0.50      0.29      0.36         7
                   OSHA 1910.157(c)(1)       0.80      0.40      0.53        10
                   OSHA 1910.303(e)(1)       1.00      0.17      0.29         6
                   OSHA 1910.303(g)(1)       0.33      0.40      0.36         5

                             micro avg       0.74      0.73      0.73        70
                             macro avg       0.73      0.54      0.57        70
                          weighted avg       0.75      0.73      0.69        70
```


Accuracy: 0.726027397260274
Precision: 0.7422945205479452
Recall: 0.726027397260274
F1 Score: 0.689552134677034
AUC: 0.9506145304088361

![Image](https://github.com/user-attachments/assets/802d9601-92c6-4195-a630-43fb9b5f45b4)
